### ISSUE TITLE
Request is inconsistent with PSR ServerRequestInterface

### ISSUE THREAD

**Issue by SilverFire** [yiisoft/yii2#15790]
1. Send a GET request:

```
GET http://test.local/test
```

2. Call `yii\web\Request::getParsedBody()`

Expected: `null` [according to PSR](https://github.com/php-fig/http-message/blob/master/src/ServerRequestInterface.php#L161)

Actual: Exception thrown [here](https://github.com/yiisoft/yii2/blob/2.1/framework/web/Request.php#L722)

**Comment by klimov-paul** [comment]
The exception your refer indicates that request should have a body according to RFC, e.g. its content type header, but it does not hold any content.
Perhaps, there must be an extra check for empty body somewhere, perhaps not - that is hard to say at the first glance.

**Comment by klimov-paul** [comment]
Confirmed. This is definitely a bug to be fixed.
At the present state following common controller code throws an exception:

```php
public function actionCreate()
{
    $model = new Foo();
    if ($model->load(Yii::$app->request->post())) { // throws `UnsupportedMediaTypeHttpException`
        $model->save();
    }
    ...
}
```

**Comment by klimov-paul** [comment]
Most likely bug was inttroduced during the pursuit for #14522 or #14458

**Comment by klimov-paul** [comment]
Resolved by commit 854ae7127cba009def2496075fc90c03c5699d8c
